src/proxy/inapp/inapp_full_proxy.go

```diff
@@ -0,0 +1,188 @@
+package app
+
+import (
+	"fmt"
+	"time"
+
+	bcrypto "github.com/mosaicnetworks/babble/src/crypto"
+	hg "github.com/mosaicnetworks/babble/src/hashgraph"
+	bproxy "github.com/mosaicnetworks/babble/src/proxy/babble"
+	"github.com/sirupsen/logrus"
+)
+
+//InmemFullProxy serves as an Inapp proxy for those whom
+type InmemFullProxy struct {
+	submitCh              chan []byte
+	commitCh              chan bproxy.Commit
+	stateHash             []byte
+	committedTransactions [][]byte
+	logger                *logrus.Logger
+	snapshotRequestCh     chan bproxy.SnapshotRequest
+	restoreCh             chan bproxy.RestoreRequest
+	timeout               time.Duration
+}
+
+// NewInmemFullProxy instantiate an InmemFullProxy.
+// If no logger, a new one is created
+func NewInmemFullProxy(timeout time.Duration, logger *logrus.Logger) *InmemFullProxy {
+	if logger == nil {
+		logger = logrus.New()
+		logger.Level = logrus.DebugLevel
+	}
+
+	return &InmemFullProxy{
+		submitCh:              make(chan []byte),
+		commitCh:              make(chan bproxy.Commit),
+		stateHash:             []byte{},
+		committedTransactions: [][]byte{},
+		logger:                logger,
+		snapshotRequestCh:     make(chan bproxy.SnapshotRequest),
+		restoreCh:             make(chan bproxy.RestoreRequest),
+		timeout:               timeout,
+	}
+}
+
+func (p *InmemFullProxy) commit(block hg.Block) []byte {
```

Comment:
I don't know if this is relevant anymore

Reply:
Yes it is not necessary anymore